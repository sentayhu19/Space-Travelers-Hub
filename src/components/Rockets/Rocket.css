.main-rocket {
  display: flex;
  gap: 20px;
  margin-top: 20px;
  margin-bottom: 20px;
}

.rocket-img {
  width: 30%;
}

.main-img {
  width: 100%;
}

.rocket-info {
  width: 70%;
}

.rocket-name {
  margin-top: 0;
}

.rocket-action {
  background: #007bff;
  border: none;
  color: #fff;
  padding: 10px;
  border-radius: 3px;
  font-size: 16px;
}

.rocket-action-cancel {
  background: #fff;
  border: 1px solid grey;
  color: grey;
  padding: 10px;
  border-radius: 3px;
  font-size: 16px;
}
