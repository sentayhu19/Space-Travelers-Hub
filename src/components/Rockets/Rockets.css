.rockets-list {
  margin-right: 50px;
  margin-left: 51px;
}
