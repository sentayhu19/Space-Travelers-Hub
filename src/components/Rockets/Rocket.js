import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { addReservation, deleteReservation } from '../../redux/rockets/action';
import './Rocket.css';

const Rocket = ({
  id, desc, name, img, reserv, i,
}) => {
  const dispatchAction = useDispatch();
  const reserve = () => dispatchAction(addReservation(id));
  const cancel = () => dispatchAction(deleteReservation(id));

  const reserveBtn = (<button type="button" id="reserveBtn" onClick={reserve} className="rocket-action">Reserve Rocket</button>);
  const deleteBtn = (<button type="button" id="cancelBtn" onClick={cancel} className="rocket-action-cancel">Cancel Reservation</button>);
  const reserved = (<mark id="reserved">Reserved</mark>);
  if (i < 4) {
    return (
      <div className="main-rocket" id={id}>
        <div className="rocket-img">
          <img src={img} alt={name} className="main-img" />
        </div>
        <div className="rocket-info">
          <h3 className="rocket-name">{name}</h3>
          <p className="rocket-desc">
            {(reserv) ? reserved : null }
            {' '}
            {desc}
          </p>
          {(reserv === false) ? reserveBtn : deleteBtn }
        </div>
      </div>
    );
  }
  return (<p />);
};

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  reserv: PropTypes.bool.isRequired,
  i: PropTypes.number.isRequired,
};

export default Rocket;
