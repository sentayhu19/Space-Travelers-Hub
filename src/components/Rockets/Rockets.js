import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { generate } from 'randomized-string';
import Rocket from './Rocket';
import { getRocketList } from '../../redux/rockets/reducers';
import './Rockets.css';

const Rockets = () => {
  const dispatchAction = useDispatch();
  const rockets = useSelector((state) => state.rocketsReducer);
  useEffect(() => {
    dispatchAction(getRocketList());
  }, []);
  return (
    <div className="rockets-list">
      { rockets.map((rocket, i) => (

        <Rocket
          key={generate()}
          id={rocket.id}
          name={rocket.rocket_name}
          img={rocket.flickr_images[0]}
          desc={rocket.description}
          reserv={rocket.reserv}
          i={i}
        />
      )) }
    </div>
  );
};

export default Rockets;
