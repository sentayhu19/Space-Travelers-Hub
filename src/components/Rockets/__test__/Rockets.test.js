import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import Rockets from '../Rockets';

const rockets = [
  {
    desc: 'Falcon 1 is',
    key: '1',
    id: '1',
    name: 'Falcon 1',
    img: 'https://dbkhs.net',
    reserv: false,
  },
];

describe('Test the Rockets Component', () => {
  it('rockets section renders', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets key={rockets[0].id} rocket={rockets[0]} />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
