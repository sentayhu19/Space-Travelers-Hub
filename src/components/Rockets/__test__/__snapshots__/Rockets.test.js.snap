// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Rockets Component rockets section renders 1`] = `
<div>
  <div
    class="rockets-list"
  />
</div>
`;
